Sources/Workspace/Workspace.swift

```diff
@@ -335,8 +335,15 @@ public class Workspace {
     }
 
     /// Resets the entire workspace by removing the data directory.
-    public func reset() throws {
-        try managedDependencies.reset()
+    ///
+    /// - Parameters:
+    ///     - diagnostics: The diagnostics engine that reports errors, warnings
+    ///       and notes.
+    public func reset(with diagnostics: DiagnosticsEngine) {
+        guard diagnostics.wrap({ try managedDependencies.reset() }) else {
```

Comment:
Yeah, I think we should either make these APIs throwing or return a bool perhaps.
PS: This is not related to this PR, just a general comment.